### DISCUSSION TITLE
Feature Request: Will string_agg be merged in the original order?

### DISCUSSION THREAD

**Discussion by DAAworld** [duckdb/duckdb#8447] [Ideas]
import pandas as pd
import duckdb

df = pd.DataFrame({"a": [0,0,1,1,2,2,3,3,4,4,5,5], "b": [1,1,1,1,1,1,1,1,1,1,1,1]})
with duckdb.connect() as con:
    con.register("df", df)
    aa = con.execute("""select "b", string_agg("a") as v from df group by "b" """).df()

Using these codes, sometimes the result is 0,0,1,1,2,2,3,3,4,4,5,5. Sometimes it is 4, 4, 5, 5, 0, 0, 1, 1, 2, 2, 3, 3. But among all the results, only this one changes repeatedly, and the others are merged in order .

**Comment by Tishj** [comment]
I think these results are undefined if you don't specify an ORDER BY clause, no?

**Reply by DAAworld** [reply]
Yes, the order of other results of groupby is stable, only this set of data changes between those two values. If I add the order by clause, all the results are also stable.

**Comment by l1t1** [comment]
```
D with t(a,b) as(values('1',1),('0',3))select string_agg(a,'|' order by b desc )from t;
┌────────────────────────────────────┐
│ string_agg(a, '|' ORDER BY b DESC) │
│              varchar               │
├────────────────────────────────────┤
│ 0|1                                │
└────────────────────────────────────┘
```
